main.cpp

```diff
@@ -205,292 +224,185 @@ struct Microwave
-    5)  length of queue at security
-3 things it can do:
-    1)  board passengers on planes
-    2)  restock coffee supplies
-    3)  stop passengers entering airport
- */
+void Microwave::cookPotato(Microwave::Preset potato)
+{
+    power = potato.power;
+    secondsRemaining = potato.seconds;
+}
+
+void Microwave::cookByWeight(double selectedWeight, double selectedPower)
+{
+    power = selectedPower;
+    secondsRemaining = selectedWeight / selectedPower;
+}
+
 
 struct Airport
 {
-    //number of planes
     int numPlanes = 50;
-    //number of passengers
     int numPassengers = 1000;
-    //number of cafes
     int numCafes = 10;
-    //stores number of coffee sales per day
-    int numCoffeeSales = 500;
-    //length of queue at security
+    int coffeeStock = 500;
     int averageMinutesInQueue = 120;
 
-    //board passengers on planes
     void boardPassengers(bool planeIsReady, int numPassengers);
-    //restock coffee supplies
-    int restockCoffee(bool coffeeHasRunOut);
-    //stop passengers entering airport
+    bool coffeeRestockNeeded(int coffeeSales);
     bool passengersCanEnter(int numPassengers, int averageMinutesInQueue);
 };
 
-/*
-Thing 4) Vinyl record
-5 properties:
-    1) Length of music on side A
-    2) Has music on both sides
-    3) Signal to noise ratio
-    4) Can play without skipping
-    5) Number of good tracks on record
-3 things it can do:
-    1) Play through to end
-    2) Bring joy to the listener
-    3) Accrue resale value on Discogs
- */
+bool Airport::coffeeRestockNeeded(int coffeeSales)
+{
+    if (coffeeStock - coffeeSales <= 50)
+    {
+        return true;
+    } 
+    else
+    {
+        return true;
+    }
+}
+
 
 struct Vinyl
 {
-    //Length of music on side A
     int lenMusicSeconds = 400;
-    //Has music on both sides
     bool musicBothSides = true;
-    //Signal to noise ratio
     double noise = 0.0;
-    //Can play without skipping
     bool vinylCanPlay = false;
-    //Number of good tracks on record
     int numberOfGoodTracks = 1;
 
-    //Replay from begining
     void replay(bool hasPlayedToEnd, bool is12Inch);
-    //Bring joy to the listener
+    void moveToneArm(bool is12Inch);
     double bringJoy(int numberOfGoodTracks, double listenerMood);
-    //Accrue resale value on Discogs
     double accrueValue(int numberOfGoodTracks, int playsOnYoutube);
 };
 
-/*
-Thing 5) Oscillator
-5 properties:
-    1) Saw Wave level
-    2) Square Wave level
-    3) Tuning control
-    4) Sub Oscillator on/off 
-    5) Octave selector
-3 things it can do:
-    1) Change octave
-    2) Add sub oscillator
-    3) Detune
- */
+void Vinyl::replay(bool hasPlayedToEnd, bool is12Inch)
+{
+    if (hasPlayedToEnd == true)
+    {
+        moveToneArm(is12Inch);
+    }
+}
+
+void Vinyl::moveToneArm(bool is12Inch)
+{
+    float distanceToMoveArm;
+
+    if (is12Inch == true)
+    {
+        distanceToMoveArm = 6.0f;
+    }
+    else
+    {
+        distanceToMoveArm = 3.5f;
+    }
```

Comment:
You can use a ternary expression here: 
```
distanceToMoveArm = (is12Inch ? 6.0f : 3.5f);
```
It's called the "Conditional Operator" and you can learn about it here: https://en.cppreference.com/w/cpp/language/operator_other
- [ ] revise